pathfinding_service.rst

```diff
@@ -219,6 +223,47 @@ Example
         "errors": "No suitable path found."
     }
 
+
+``GET api/v1/<token_network_address>/payment/iou``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Request the last IOU used by ``sender`` to pay the PFS.
+This IOU can be used by the client to generate the next IOU to pay the PFS by increasing the ``amount`` and updating the signature.
+
+Arguments
+"""""""""
+
++---------------------+------------+---------------------------------------------------------+
+| Field Name          | Field Type | Description                                             |
++=====================+============+=========================================================+
+| sender              | address    | Sender of the payment (Ethereum address of client)      |
++---------------------+------------+---------------------------------------------------------+
+| receiver            | address    | Receiver of the payment (Ethereum address of PFS)       |
++---------------------+------------+---------------------------------------------------------+
+| timestamp           | string     | Current UTC date and time in ISO 8601 format            |
+|                     |            | (e.g. 2019-02-25T12:53:16Z)                             |
++---------------------+------------+---------------------------------------------------------+
+| signature           | bytes      | Signature over the other three arguments [#sig]_        |
++---------------------+------------+---------------------------------------------------------+
+
+.. [#sig] The signature is calculated by
+          ::
+
+               ecdsa_recoverable(privkey,
+                                 sha3_keccak("\x19Ethereum Signed Message:\n[LENGTH]"
+                                             || sender || receiver || timestamp ))
+
+Returns
+"""""""
+A JSON object with a single property:
+
++----------------------+---------------+-----------------------------------------------+
+| Field Name           | Field Type    | Description                                   |
++======================+===============+===============================================+
+| last_iou             | object        | IOU object as described in :ref:`pfs_payment` |
```

Comment:
So this would be:
```
{
   "last_iou": {
        "fields1": ...,
        "fields2": ...
    }
}
```
Wouldn't be easier to just return the IOU object itself without nesting?

Reply:
Yes, but this way we can easily add new fields like `last_usage_time` or `unclaimed_ious` without breaking compatibility or adding new endpoints.

Reply:
I see... good point.